galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RequestorRas.java

```diff
@@ -70,49 +61,56 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
        
+    	
+        try {
+        	
     	//gets string query as hashmap
+        
     	Map<String, String[]> query = req.getParameterMap();
-    	
+
     	//gets requestors
-    	List<String> list = getRequestors(resp);
+    	List<String> list = getRequestors();
     	
     	//sorts list
     	Collections.sort(list);
     	
-    	JsonObject requestors = new JsonObject();
-    	
-    	Gson gson = new Gson();    	
-    	
-    	if(!ExtractQuerySort.isAscending(query, "requestor")) {
-    		Collections.reverse(list);
-    	}
+    	JsonObject requestors = new JsonObject();   	
     	
+    	if(!query.isEmpty()) {
+		    if(!ExtractQuerySort.isAscending(query, "requestor")) {
+		    		Collections.reverse(list);
+		    }
+        }
+    
     	//create json object
     	JsonElement json = new Gson().toJsonTree(list);
     	requestors.add("requestors", json);
-    	
-        try {
-            
+        
         PrintWriter out = resp.getWriter();
         resp.setHeader("Content-Type", "Application/json");
+        resp.setCharacterEncoding("UTF-8");
         out.print(requestors);
         out.close();
         
         }
         catch(Exception e) {
-            e.printStackTrace();
+            throw new ServletException();
```

Comment:
this will destroy the stacktrace.
throw new ServletException("Error occurred during get requestors",e);

This will give an indication of the problem, most importantly will give the full stracktrace so you can figure out where the original problem occurred